File: backend/src/domain/models/document.py
"""Document ORM model — an AI-generated note from chat conversations."""

import uuid

from sqlalchemy import Boolean, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from src.domain.models.base import Base, TimestampMixin, UUIDMixin


class Document(Base, UUIDMixin, TimestampMixin):
    __tablename__ = "documents"

    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    preview: Mapped[str] = mapped_column(String(500), default="", nullable=False)
    content: Mapped[str] = mapped_column(Text, default="", nullable=False)
    tag: Mapped[str] = mapped_column(
        String(20), default="学习", nullable=False
    )
    is_pinned: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    status: Mapped[str] = mapped_column(
        String(20), default="ready", nullable=False
    )  # "generating" | "ready" | "failed"

    def __repr__(self) -> str:
        return f"<Document id={self.id!r} title={self.title!r}>"

File: app/src/types/document.ts
/** 笔记/文档相关类型 */

export type NoteTag = '学习' | '工作' | '想法' | '收藏';

export type NoteStatus = 'generating' | 'ready' | 'failed';

export interface Note {
  id: string;
  title: string;
  preview: string;
  tag: NoteTag;
  isPinned: boolean;
  content: string;
  status: NoteStatus;
  created_at: string;
  updated_at: string;
}

export interface CreateNoteRequest {
  title: string;
  content?: string;
  tag?: NoteTag;
}
